std/algorithm/sorting.d

```diff
@@ -2889,18 +2889,46 @@ void partialSort(alias less = "a < b", SwapStrategy ss = SwapStrategy.unstable,
     Range)(Range r, size_t n)
     if (isRandomAccessRange!(Range) && hasLength!(Range) && hasSlicing!(Range))
 {
-    topN!(less, ss)(r, n);
-    sort!(less, ss)(r[0 .. n]);
+    partialSort!(less, ss)(r[0 .. n], r[n .. $]);
 }
 
 ///
-@safe unittest
```

Comment:
If we mark a generic function as `@trusted`, that includes whatever operations it uses of the generic type received as a parameter, which creates a safety hole (you may do arbitrary unsafe operations in a type and instantiate the function with it).

To make a generic function `@safe`, don't use `@safe` nor `@trusted` with it. Instead, use only safe operations in the function and let deduction do the rest for you.

If you figure you need to do something trusted, use small `@trusted` lambdas like in https://github.com/dlang/druntime/pull/1710. All calls into generic/unknown code should be outside those `@trusted` portions.